packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -406,6 +416,29 @@ - (void)updatePlayingState {
   _displayLink.running = _isPlaying || self.waitingForFrame;
 }
 
+- (void)updateRate {
+  // See https://developer.apple.com/library/archive/qa/qa1772/_index.html for an explanation of
+  // these checks.
+  // If status is not AVPlayerItemStatusReadyToPlay then both canPlayFastForward
+  // and canPlaySlowForward are always false and it is unknown whether video can
+  // be played at these speeds, updatePlayingState will be called again when
+  // status changes to AVPlayerItemStatusReadyToPlay.
+  float speed = _playbackSpeed.floatValue;
+  if (speed > 2.0 && !_player.currentItem.canPlayFastForward) {
+    if (_player.currentItem.status != AVPlayerItemStatusReadyToPlay) {
```

Comment:
what if we call `updateRate` with speed < 2.0? should we also check the status here? Should this status check be moved to top of the fucntion?

Reply:
That check is to confirm whether `canPlay...` returned false because it cannot be played at speed>2 instead of because status was not ready to play. There is no need to check status for speeds for which is not needed `canPlay...`. But as I am thinking about it now maybe status should be fetched before `canPlay...` to avoid a false positive when status becomes ready right after `canPlay...`.